### PR TITLE
dev-guide: add limit.md

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

**Comment by xiang90** [comment]
Can you add a dedicated limit sections under https://github.com/coreos/etcd/blob/master/Documentation/docs.md#developing-with-etcd?

I do not want to put all stuff to lose structured faq, until one question becomes really popular in github issues.

File: Documentation/faq.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 ### etcd, general
 
+#### When to use etcd
```
Comment (xiang90) [inline Documentation/faq.md]: this should be in the why etcd section that @fanminshi is working on now. do not put this in faq

File: Documentation/faq.md [hunk 2]
```diff
@@ -2,6 +2,14 @@
...
+
+etcd is designed to store a small amount of metadata in consistent, fault-tolerant way. Distributed systems use etcd as the root of scheduling and service discovery configuration storage. [Distributed lock][etcd-concurrency] is implemented on top of etcd, useful for master election in other distributed databases.
+
+#### When not to use etcd (etcd constraints)
+
+etcd can only handle small chunks data: configuration file, JSON, YAML, etc. etcd limits the size of request in 1.5MB; see [`maxRequestBytes`][max-request-limit]. And default storage size limit is 2GB, configurable with `--quota-backend-bytes` flag; supports up to 8GB, see [`DefaultQuotaBytes`][max-storage-limit].
```
Comment (fanminshi) [inline Documentation/faq.md]: `etcd can only handle small chunks data` => `etcd can only handle small chunks of data`

File: Documentation/faq.md [hunk 3]
```diff
@@ -98,6 +106,9 @@ Every new etcd cluster generates a new cluster ID based on the initial cluster c
 
 Usually this warning happens after tearing down an old cluster, then reusing some of the peer addresses for the new cluster. If any etcd process from the old cluster is still running it will try to contact the new cluster. The new cluster will recognize a cluster ID mismatch, then ignore the request and emit this warning. This warning is often cleared by ensuring peer addresses among distinct clusters are disjoint.
 
+[etcd-concurrency]: https://godoc.org/github.com/coreos/etcd/clientv3/concurrency
+[max-request-limit]: https://github.com/coreos/etcd/blob/master/etcdserver/v3_server.go#L40-L44
```
Comment (xiang90) [inline Documentation/faq.md]: let's remove the reference to code. doc itself is ok.

File: Documentation/faq.md (hunk 2 above)
Comment (fanminshi) [inline Documentation/faq.md]: `etcd limits the size of request in 1.5MB` => `etcd limits the size of request to 1.5MB`

**Comment by gyuho** [comment]
@xiang90 Just moved to limit.md. PTAL. Thanks.

File: Documentation/dev-guide/limit.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Request Size Limitation
+
+etcd can only handle small chunks of data: configuration file, JSON, YAML, etc. etcd limits the size of request to 1.5MB.
+
+# Storage Size Limitation
```
Comment (xiang90) [inline Documentation/dev-guide/limit.md]: Store size limit

File: Documentation/dev-guide/limit.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Request Size Limitation
```
Comment (xiang90) [inline Documentation/dev-guide/limit.md]: we should link this doc from doc.md at the top level.

Comment (gyuho) [inline Documentation/dev-guide/limit.md]: All done. Thanks.

**Comment by xiang90** [comment]
defer to @fanminshi and @heyitsanthony

File: Documentation/dev-guide/limit.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Request size limit
```
Comment (heyitsanthony) [inline Documentation/dev-guide/limit.md]:
```
# etcd system limits

## Request size limit
```
?

File: Documentation/dev-guide/limit.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+
+The size limit is 1MB. etcd is designed to handle small key value paris. Very large request might block other requests for a long time. This also applies to `Txn` requests. We might loosen the size in the future a little bit or make it configurable.
```
Comment (fanminshi) [inline Documentation/dev-guide/limit.md]: s/paris/pairs/

Comment (heyitsanthony) [inline Documentation/dev-guide/limit.md]: etcd is designed to handle small key value pairs typical for metadata. Larger requests will work, but may increase the latency of other requests. For the time being, etcd guarantees to support RPC requests with up to 1MB of data. In the future, the size limit may be loosened or made it configurable.

File: Documentation/dev-guide/limit.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+
+# Storage size limit
+
+Default storage size limit is 2GB, configurable with `--quota-backend-bytes` flag; supports up to 8GB.
```
Comment (heyitsanthony) [inline Documentation/dev-guide/limit.md]: s/Default/The default/

File: Documentation/docs.md [hunk 9]
```diff
@@ -17,6 +17,7 @@ The easiest way to get started using etcd as a distributed key-value store is to
  - [gRPC naming and discovery][grpc_naming]
  - [Embedding etcd][embed_etcd]
  - [Experimental features and APIs][experimental]
+ - [Limit][limit]
```
Comment (heyitsanthony) [inline Documentation/docs.md]: System limits?

**Comment by gyuho** [comment]
@heyitsanthony @fanminshi All fixed. PTAL. Thanks!

File: Documentation/dev-guide/limit.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+# etcd system limits
```
Comment (xiang90) [inline Documentation/dev-guide/limit.md]: remove etcd? Just System limits?

**Comment by heyitsanthony** [comment]
lgtm after addressing xiang's comment

**Comment by gyuho** [comment]
Fixed. Merging. Thanks!

**Pull request merged**